dds/DCPS/transport/rtps_udp/RtpsUdpDataLink.cpp

```diff
@@ -3698,6 +3705,11 @@ RtpsUdpDataLink::RtpsWriter::gather_nack_replies_i(MetaSubmessageVec& meta_subme
   }
 
   {
+    for (FragmentInfo::const_iterator pos = consolidated_fragment_requests.begin(),
+           limit = consolidated_fragment_requests.end(); pos != limit; ++pos) {
+      consolidated_requests.erase(pos->first);
+    }
+
```

Comment:
Could this be removed?  May not be needed if the NackFrags are only generated when the sequence number is not nacked.

Reply:
I'll give it a try and test it locally